### PR TITLE
Tracking doc: rewrite `@wdio/mocha-framework` package into TypeScript

### PR DESCRIPTION
**Pull request by christian-bromann**
This is a tracking issue for organising the effort of re-writing the code base from vanilla JS into TypeScript. If you are reading this and are interested getting involved, this is a great chance to do so. Here are the rules:

- if you are interested converting source files from `/packages/wdio-mocha-framework` into TypeScript please comment on this thread and mention the files you want to convert to avoid duplicate work
- there is no reason to convert the whole package at once - smaller PRs make it easier to review the code
- if you make a PR, reference this issue so we can keep track of the progress
- if you make a PR, label it with `TypeScript` and the package name `wdio-mocha-framework` so we can easier find all PRs
- if you make a PR, assign it to the `TypeScript Rewrite` project

Happy rewriting 😅

### CODE REVIEW

File: packages/wdio-mocha-framework/src/utils.ts [hunk 1]
```diff
@@ -1,5 +1,8 @@
-export function loadModule (name, context) {
+import type { MochaContext } from './types'
+
+export function loadModule (name: string, context: MochaContext) {
     try {
+        // @ts-ignore
         module.context = context
```
Comment (mgrybyk) [inline packages/wdio-mocha-framework/src/utils.ts:6]: [_not relations to the PR_] Looks like some dark magic :)

**Review by mgrybyk — CHANGES_REQUESTED** [review]

File: packages/wdio-mocha-framework/src/index.ts [hunk 2]
```diff
@@ -26,32 +28,37 @@ const DEFAULT_INTERFACE_TYPE = 'bdd'
...
-        this.specs = specs
-        this.config = Object.assign({
+    private _mocha?: Mocha
+    private _runner?: Runner
+    private _specLoadError?: Error
+
+    private _level = 0
+    private _hasTests = true
+    private _suiteIds: string[] = ['0']
+    private _suiteCnt: Map<string, number> = new Map()
```
[the remark this answers is not in the document]
Comment (christian-bromann) [reply packages/wdio-mocha-framework/src/index.ts:41]: I looked into it and apparently this is desired for the same reason we defined `_suiteIds` as string (see e.g. [here](https://github.com/webdriverio/webdriverio/blob/master/packages/wdio-mocha-framework/src/index.js#L324)). How this plays out you can see in [this test](https://github.com/webdriverio/webdriverio/blob/master/packages/wdio-mocha-framework/tests/adapter.test.js#L360-L412). Definitely worth improving eventually but not as part of this PR.

File: packages/wdio-mocha-framework/src/utils.ts (hunk 1 above)
Comment (christian-bromann) [reply packages/wdio-mocha-framework/src/utils.ts:6]: true, let's keep it as is and eventually clean this up.

Comment (mgrybyk) [reply packages/wdio-mocha-framework/src/utils.ts:6]: agree, no idea how it works

**Review by mgrybyk — APPROVED** [review]

**Pull request merged**